### PR TITLE
Add cypress

### PR DESCRIPTION
**Pull request by skaidra-bake**
Hi, I have been learning about Cypress lately and it sounds like a really cool testing framework, which works very fast and works in the context of the app! It could be interesting to replace the current system tests in the future so the speed and accuracy could be improved.

Only with an initial test for signing in, I found a couple of problems with the devise initializer and it's interaction with turbo-rails and alerts styling (they weren't visible). It's WIP of course, but I believe the changes are quite useful. What do you think?

### CODE REVIEW

File: spec/cypress/app_commands/activerecord_fixtures.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  logger.debug "loading fixtures: { dir: #{fixtures_dir}, files: #{fixture_files} }"
+  ActiveRecord::FixtureSet.reset_cache
+  ActiveRecord::FixtureSet.create_fixtures(fixtures_dir, fixture_files)
+  "Fixtures Done" # this gets returned
+else # this else part can be removed
+  logger.error "Looks like activerecord_fixtures has to be modified to suite your need"
+  Post.create(title: 'MyCypressFixtures')
+  Post.create(title: 'MyCypressFixtures2')
+  Post.create(title: 'MyRailsFixtures')
+  Post.create(title: 'MyRailsFixtures2')
```
Comment (denissellu) [inline spec/cypress/app_commands/activerecord_fixtures.rb]: @skaidra-bake lets remove this part as we are using active-record😄

File: spec/cypress/app_commands/clean.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+if defined?(DatabaseCleaner)
+  # cleaning the database using database_cleaner
+  DatabaseCleaner.strategy = :truncation
+  DatabaseCleaner.clean
+else
+  logger.warn "add database_cleaner or update cypress/app_commands/clean.rb"
+  Post.delete_all if defined?(Post)
```
Comment (denissellu) [inline spec/cypress/app_commands/clean.rb:7]: @skaidra-bake we have database cleaner, so we can also remove this

Comment (denissellu) [inline spec/cypress/app_commands/clean.rb]: @skaidra-bake lets remove that line and just keep the `logger`

File: spec/cypress/app_commands/scenarios/basic.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# You can setup your Rails state here
+# MyModel.create name: 'something'
+Post.create(title: 'I am a Postman')
```
Comment (denissellu) [inline spec/cypress/app_commands/scenarios/basic.rb]: @skaidra-bake lets I think we can delete this as we are not yet ready to test this

File: spec/cypress/fixtures/example.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
```
Comment (denissellu) [inline spec/cypress/fixtures/example.json]: We can also remove this file

File: spec/cypress/integration/rails_examples/other_spec.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+describe('Rails Other examples', function() {
+  it('cypress eval', function() {
+    cy.app('clean') // have a look at cypress/app_commands/clean.rb
+    cy.appEval("Post.create(title: 'Hello Eval')")
```
Comment (denissellu) [inline spec/cypress/integration/rails_examples/other_spec.js]: Lets remove this line as well

File: spec/cypress/integration/rails_examples/other_spec.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    cy.get('table').find('tbody').should(($tbody) => {
+      expect($tbody).not.to.contain('Multi Command')
+      expect($tbody).to.contain('Hello Eval')
+    })
+  })
+
+  it('runs multiple commands', function() {
+    cy.appCommands([{ name: 'clean' },
+                    { name: 'scenarios/basic' },
+                    { name: 'eval', options: "Post.create(title: 'Multi Command')" }])
```
Comment (denissellu) [inline spec/cypress/integration/rails_examples/other_spec.js]: Remove this as well

```
cy.appCommands([{ name: 'clean' },
                    { name: 'scenarios/basic' }])
```

File: spec/cypress/app_commands/scenarios/basic.rb (hunk 3 above)
Comment (skaidra-bake) [reply spec/cypress/app_commands/scenarios/basic.rb]: Oh yes, this does not exist anymore I don't think

**Comment by denissellu** [comment]
🤔  I think we can remove the `spec/cypress/integration/rails_examples/` folder, as it is just an example to show us how to do cypress with rails

File: spec/cypress/support/on-rails.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+Cypress.Commands.add('appFixtures', function (options) {
+  cy.app('activerecord_fixtures', options)
+});
+// CypressOnRails: end
+
+// The next is optional
+// beforeEach(() => {
+//  cy.app('clean') // have a look at cypress/app_commands/clean.rb
+// });
```
Comment (denissellu) [inline spec/cypress/support/on-rails.js]: and lets clean this up

File: config/initializers/devise.rb [hunk 8]
```diff
@@ -296,7 +296,7 @@ def to_turbo_stream
   # should add them to the navigational formats lists.
   #
   # The "*/*" below is required to match Internet Explorer requests.
-  # config.navigational_formats = ['*/*', :html]
+  config.navigational_formats = ['*/*', :html, :turbo_stream]
```
Comment (denissellu) [inline config/initializers/devise.rb:299]: !!! great hunting and spotting! 🔥

File: spec/cypress/support/on-rails.js [hunk 9]
```diff
@@ -22,7 +22,7 @@ Cypress.Commands.add('app', function (name, command_options) {
   });
 });
 
-Cypress.Commands.add('appScenario', function (name, options = {}) {
+Cypress.Commands.add('scenario', function (name, options = {}) {
```
Comment (denissellu) [inline spec/cypress/support/on-rails.js:25]: Nice! I think this read a bit better

File: spec/cypress/app_commands/clean.rb (hunk 2 above)
Comment (skaidra-bake) [reply spec/cypress/app_commands/clean.rb]: @denissellu I'm not sure about this. which lines should be kept and which should be deleted?

File: package.json [hunk 10]
```diff
@@ -3,7 +3,7 @@
     "private": true,
     "scripts": {
         "cy:open": "cypress open --project ./spec",
-        "cy:server": "bundle exec rails server -e test -p 5002 -P 69"
+        "cy:server": "bundle exec rails server -e test -p 5002 -P tmp/pids/cypress.pid"
```
Comment (denissellu) [inline package.json:6]: 🚀

**Comment by skaidra-bake** [comment]
@denissellu  I think this is ready

**Review by denissellu — APPROVED** [review]
Good work

**Pull request merged**